src/NPZ.jl

```diff
@@ -58,36 +57,37 @@ end
 # be fixed by rehashing the Dict when the module is
 # loaded.
 
-readle(ios::IO, ::Type{UInt16}) = htol(read(ios, UInt16))
+readle(ios::IO, ::Type{T}) where T = ltoh(read(ios, T)) #ltoh is inverse of htol
 
-function writele(ios::IO, x::Vector{UInt8})
-    n = write(ios, x)
-    if n != length(x)
-        error("short write")
-    end
-    n
+function writecheck(io::IO, x::Any)
+    n = write(io, x) # returns size in bytes
+    n == sizeof(x) || error("short write") # sizeof is size in bytes
 end
 
-writele(ios::IO, x::String) = writele(ios, convert(Vector{UInt8}, x))
-writele(ios::IO, x::UInt16) = writele(ios, reinterpret(UInt8, [htol(x)]))
+# Endianness only pertains to multi-byte things
+writele(ios::IO, x::AbstractVector{UInt8}) = writecheck(ios, x)
+# NPY headers only have ascii strings (bytes) so endianness doesn't matter
```

Comment:
I don't think you should be assuming we're writing NPY header. In the future, we might be writing string data. Either change it back take a `String` input instead of `AbstractString` so that the input is UTF-8 encoded or handle byte order for any type of string (e.g. UTF-16).

In fact, I don't understand why all the `String` needs to be converted to `AbstractString` in the functions below. Since the header is ASCII, `String` works for us. There is no need to be more general, which has potential to slow down parsing.